examples/instrumentation-quickstart/integration_test/instrumentation_test.go

```diff
@@ -0,0 +1,154 @@
+}
+
+func TestApp(t *testing.T) {
+	ctx := context.Background()
+	composeStack := composeUp(ctx, t)
+
+	// Let the docker compose app run until some spans/logs/metrics are sent to GCP
+	t.Logf("Compose stack is up, waiting for prometheus metrics indicating successful export")
+
+	// Check the collector's self-observability prometheus metrics to see that exports to GCP were successful.
+	for _, tc := range []testCase{
+		{metricName: "otelcol_exporter_sent_spans", exporter: "googlecloud"},
+		{metricName: "otelcol_exporter_sent_log_records", exporter: "googlecloud"},
+		{metricName: "otelcol_exporter_sent_metric_points", exporter: "googlemanagedprometheus"},
+	} {
+		t.Run(tc.metricName, func(t *testing.T) {
+			require.EventuallyWithT(
+				t,
+				func(collect *assert.CollectT) {
+					promMetrics, err := getPromMetrics(ctx, composeStack)
+					if !assert.NoError(collect, err) {
+						return
+					}
+					verifyPromMetric(collect, promMetrics, tc)
+				},
+				time.Minute*2, // wait for up to
+				time.Second,   // check at interval
+			)
+		})
+	}
+}
+
+func composeUp(ctx context.Context, t *testing.T) compose.ComposeStack {
+	composeFiles := []string{"../docker-compose.yaml"}
+	if composeOverrideFile := os.Getenv("COMPOSE_OVERRIDE_FILE"); composeOverrideFile != "" {
+		composeFiles = append(composeFiles, strings.Split(composeOverrideFile, ",")...)
+	}
+
+	var (
+		composeStack compose.ComposeStack
+		err          error
+	)
+	composeStack, err = compose.NewDockerCompose(composeFiles...)
+	require.NoError(t, err)
+
+	require.NoError(t, err)
+	composeStack = composeStack.WithOsEnv().
+		WaitForService("app", wait.ForHTTP("/single").WithPort("8080")).
+		WaitForService("otelcol", wait.ForHTTP("/metrics").WithPort("8888"))
+
+	t.Cleanup(func() {
+		require.NoError(t, composeStack.Down(ctx, compose.RemoveOrphans(true)))
+	})
+	require.NoError(t, composeStack.Up(ctx))
+	return composeStack
+}
+
+func getPromMetrics(ctx context.Context, composeStack compose.ComposeStack) (map[string]*dto.MetricFamily, error) {
+	promUri, err := getPromEndpoint(ctx, composeStack)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := http.Get(promUri)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	var parser expfmt.TextParser
+	parsed, err := parser.TextToMetricFamilies(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+	return parsed, nil
+}
+
+func getPromEndpoint(ctx context.Context, composeStack compose.ComposeStack) (string, error) {
+	collectorContainer, err := composeStack.ServiceContainer(ctx, "otelcol")
+	if err != nil {
+		return "", err
+	}
+	collectorHost, err := collectorContainer.Host(ctx)
+	if err != nil {
+		return "", err
+	}
+	collectorPort, err := collectorContainer.MappedPort(ctx, "8888")
+	if err != nil {
+		return "", err
+	}
+	return fmt.Sprintf("http://%s:%s/metrics", collectorHost, collectorPort.Port()), nil
+}
+
+func verifyPromMetric(t assert.TestingT, promMetrics map[string]*dto.MetricFamily, tc testCase) {
```

Comment:
@dashpole is there a better way to make assertions against prometheus text format? These seem to be protobuf structs which are not very ergonomic and possibly deprecated. I could do regex like we did for Java, but seems a little fragile.

Reply:
Consider using [testutil.ScrapeAndCompare](https://github.com/prometheus/client_golang/blob/97aa0493ebfa75f20fe51ebb32f1e5c5315e5fa1/prometheus/testutil/testutil.go#L162).  Its usage is similar to GatherAndCompare, which we use to test the otel-go prometheus exporter: https://github.com/open-telemetry/opentelemetry-go/blob/48fedfa58e35b5f3b08fe54e45cfad441dd82426/exporters/prometheus/exporter_test.go#L449

Reply:
It looks like this requires exact numeric values, so I'm not sure I can get it working in an un-flaky way. I'll look through the other testutil though, thanks!